### ISSUE TITLE
Calling alGetError after dynamically loading Win64/soft_aol.dll procs returns 0xA004 (AL_INVALID_OPERATION)

### ISSUE THREAD

**Issue by ukaer** [kcat/openal-soft#978]
If this error is ignored, the loaded dll still works as expected and audio works - This includes subsequent calls to alGetError.

Note that:
- Calling alGetError multiple times one after the other keeps returning the same 0xA004 (i.e. the error is not cleared).
- Calling alGetString on this first error (0xA004) also returns null.

This behavior is not seen the non-software OpenAL implementation.

Basically code does:

HMODULE openal_dll = LoadLibraryA("../libs/soft_oal.dll");
// Check openal_dll is not NULL

// For each function needed:
... = GetProcAddress(openal_dll, ...);
// verify not NULL

ALenum al_error = alGetError(); // Returns 0xA004

**Comment by kcat** [comment]
`alGetError`, like the other al* functions, uses the current context. It's not valid to call it without a current context, so OpenAL Soft indicates that by returning `AL_INVALID_OPERATION` (it can't do the requested operation as there's no context to get the error from). Essentially, the `alGetError` call itself is generating the error when called without a current context.

Other implementations returns `0` from any function that returns a value as a default value without a context, which also happens to be `AL_NO_ERROR`. e.g.
```c++
ALuint source;
alGenSources(1, &source);
if(alGetError() == AL_NO_ERROR)
    printf("ok!\n");
```
Will print "`ok!`" if called without a context on other implementations, even though `alGenSources` will have not done anything and left `source` uninitialized.

**Comment by ukaer** [comment]
Ah gotcha, that makes a lot of sense and reading the function comment again it's clear now.

Thanks for the quick reply!
